core/src/main/scala/org/apache/spark/resource/ResourceInformation.scala

```diff
@@ -69,7 +69,7 @@ private[spark] object ResourceInformation {
    * Parses a JSON string into a [[ResourceInformation]] instance.
    */
   def parseJson(json: String): ResourceInformation = {
-    implicit val formats = DefaultFormats
+    implicit val formats: DefaultFormats.type = DefaultFormats
```

Comment:
Does it have to be `DefaultFormats.type`? Can it be `DefaultFormats` or `Formats`?

From the signature of the function `extract`, maybe it's ok to unify it as `Formats`?

```scala
 def extract[A](implicit formats: Formats, mf: scala.reflect.Manifest[A]): A =
    Extraction.extract(jv)(formats, mf)
```

Reply:
done